src/main/java/eu/europa/ted/eforms/sdk/repository/SdkCodelistRepository.java

```diff
@@ -4,38 +4,56 @@
 import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.text.MessageFormat;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
-import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import org.apache.commons.io.FilenameUtils;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import com.helger.genericode.Genericode10CodeListMarshaller;
 import com.helger.genericode.v10.CodeListDocument;
 import com.helger.genericode.v10.Identification;
 import com.helger.genericode.v10.LongName;
-import com.helger.genericode.v10.SimpleCodeList;
 import eu.europa.ted.eforms.sdk.entity.SdkCodelist;
 import eu.europa.ted.eforms.sdk.entity.SdkEntityFactory;
 import eu.europa.ted.util.GenericodeTools;
 
 public class SdkCodelistRepository extends HashMap<String, SdkCodelist> {
   private static final long serialVersionUID = 1L;
 
-  private transient Path codelistsPath;
+  private static final Logger logger = LoggerFactory.getLogger(SdkCodelistRepository.class);
+
+  private transient Path codelistsDir;
   private String sdkVersion;
 
+  private final Map<String, Path> codelistFiles;
+  private final Genericode10CodeListMarshaller marshaller;
+
   @SuppressWarnings("unused")
   private SdkCodelistRepository() {
     throw new UnsupportedOperationException();
   }
 
-  public SdkCodelistRepository(String sdkVersion, Path codelistsPath) {
+  public SdkCodelistRepository(String sdkVersion, Path codelistsDir) {
     this.sdkVersion = sdkVersion;
-    this.codelistsPath = codelistsPath;
+    this.codelistsDir = codelistsDir;
+
+    marshaller = GenericodeTools.getMarshaller();
+
+    try {
+      this.codelistFiles = getCodelistPaths(codelistsDir);
```

Comment:
Looks OK, in general I call maps in a descriptive way to explain the key value, for example:
codelistFileByCodelistId (or just codelistFileById) instead of codelistFiles
I would use the name `codelistFiles` for a simple list of codelist files but not for a map.
This naming also helps when doing a .get(xyz) so you know what the xyz is supposed to be
Just a suggestion.

Reply:
Done